### DISCUSSION TITLE
Output Schema

### DISCUSSION THREAD

**Discussion by noahmclean** [CIRDLES/Tripoli#159] [Ideas]
What do we want Tripoli to output for use in different applications (archiving, data reduction, etc)?  What type of file? What kinds of data?

**Comment by noahmclean** [comment]
Simple csv output: 
All plotted ratios: basic statistics.  
A row for each isotope ratio.  A column for 1 - mean, 2- standard error (1s absolute), 3- rejection statistics (e.g., 98/100 cycles).

**Reply by ryanickert** [reply]
This is great, and also I'd like some way to customize the output order in the csv file, and perhaps save that customization.  That's something I run into with OG Tripoli that would make my life easier.

**Reply by noahmclean** [reply]
Cool.  So, what are all the possible ingredients in a custom output file?  Isotope ratio (arithmetic, geometric) means, standard deviations, standard errors, rejection stats. Covariance matrices, probably.  What else?

**Reply by ryanickert** [reply]
Like I said below, we should have the option of including basic metadata, like the TIMS data filename, date/time of analysis etc.

**Comment by ryanickert** [comment]
I'd like the ability to output the filename that contains the original mass spec data and a date and timestamp for the beginning(?) of the analysis.

**Reply by noahmclean** [reply]
Cool, so the filename ingredients and format should be options as a part of the custom output schema? How many parts does it need, and what are they? Sample name, aliquot name, analysis name (leaving room for duplicate analyses), maybe... what else?

So I understand, you're something like NBS981_bead12_run1_2023-09-06_19-34T193904.csv? The 2023-09-06 would be the date and the 193904 would be the time, following a T, per ISO 8601. No colons are allowed in Windows filenames, so we can't format a time like 19:39:04. https://en.wikipedia.org/wiki/ISO_8601

**Reply by ryanickert** [reply]
No I just mean as part of the csv output.  So the filename could be something short and boring (mine are someting like C231) and then as part of the csv output I would have the option of including (along wiht the means SE, cov etc as you describe above) a field for the original filename (C204_U-4011.TIMSAD) and then some machine readable field for either date+time or date and then also time.

**Comment by bowring** [comment]
Please supply an example file to eliminate confusion.  Thanks for specifying this!

**Comment by noahmclean** [comment]
Here is a first stab at a minimum csv output for a single analysis.

The header contains: Sample Name, Fraction Name, Filename, Method Name, Analysis Date, Analysis Time

The (minimum!) data summary contains summary statistics for the whole analysis: intensities of species identified (here, 204Pb, 205Pb, 206Pb, 207Pb, 208Pb), isotope ratios (here, 204Pb/206Pb, 205Pb/206Pb, 207Pb/206Pb, and 208Pb/206Pb), and an analytical parameter (here, the derived Daly/Faraday gain).  

Next things to work out: 
- What is the complete list of potential header items? 
- What levels of data summary do we want? To get us started: 
  - session, 
  - fraction (multiple analyses of a single fraction), 
  - analysis, 
  - block, 
  - cycle)
- How do we build a list of parameters that can be output as rows in these data tables?
- What about instrument parameters pulled in from instrument parameter files?
- What is the complete list of statistics we want to report ?  To start: 
  - mean, 
  - standard deviation (abs or pct), 
  - standard error (abs or pct) [and what do these mean for beam-interpolated data?], 
  - the number included and total number, others?

[Redux_MinimumOutputExample.csv](https://github.com/CIRDLES/Tripoli/files/12783379/Redux_MinimumOutputExample.csv)

**Reply by bowring** [reply]
sample and fraction are not specified in data files unless they are encoded in file name?
need to define Analysis date and time
